internal/expr/expr_test.go

```diff
@@ -618,6 +692,11 @@ func (s *ExprSuite) TestValidQuery(c *C) {
 		[]any{Person{}, StringMap{}},
 		[]any{Person{ID: 666}, StringMap{"street": "Highway to Hell"}},
 		[]any{sql.Named("sqlair_0", "Highway to Hell"), sql.Named("sqlair_1", 666)},
+	}, {
+		"SELECT foo FROM t WHERE x = $string, y = $int",
+		[]any{0, ""},
+		[]any{"arg1", 2},
+		[]any{sql.Named("sqlair_0", "arg1"), sql.Named("sqlair_1", 2)},
```

Comment:
Should you be allowed to use `Named("sqlair_0", ...)` ?
That feels (to me) like it goes against the whole point of Named. Named is that you are mapping the human name present in the SQL that was supplied to an input parameter. But this suddenly changes it that "positional args also have names".
Especially what does it mean if you did:
```[]any{sql.Named("myparam", "arg1"), sql.Named("sqlair_0", 2)}```
How do you know what parameter `sqlair_0` is if it is positional, or named, or?

Reply:
So this tests the generated arguments that are designed to be passed to the function. It is part of the public api of the `expr` package but the results of this function are not exposed at all by the `sqlair` package, they are passed directly to the database.

Using named parameters in the way you suggest above is not at all supported by SQLair. The line you refer to above is the expected output of this function, it only ever generates arguments with the prefix "_sqlair_".